### PR TITLE
Added fixer for PHPUnit's @expectedException annotation

### PR DESCRIPTION
**Pull request by ro0NL**
In favor of #2200
- [x] Rename to `PhpUnitFqcnAnnotationFixer`

### CODE REVIEW

File: Symfony/CS/Fixer/Symfony/PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixer.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+final class PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens->findGivenKind(T_DOC_COMMENT) as $token) {
```
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixer.php]: don't use find for iterating. it's slower than regular loop.

Comment (ro0NL) [inline Symfony/CS/Fixer/Symfony/PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixer.php]: Revised.

**Comment by keradus** [comment]
>  Only fix classes extending \PHPUnit_Framework_TestCase

Nope, we don't do that assumption in any of our PHPUnit fixer.
`PHPUnit_Framework_TestCase` could not be used directly but custom child of it is used.
Eg we are using `PhpCsFixer\Test\AbstractFixerTestCase` in most of our tests files.

**Comment by ro0NL** [comment]
`is_subclass_of`? I mean.. we shouldnt just update every call to `setExpectedException`... right?

**Comment by keradus** [comment]
When you fix a code you have only file-level context.

If given class is defined outside of file, you have no idea what it is. It may be even defined in different repo

**Comment by ro0NL** [comment]
Yeah.. but we should verify the class/file we're updating is actually a test class/file or am i missing something..

**Comment by SpacePossum** [comment]
>Normalize setExpectedException('Some\Exception'); to '\Some\Exception'

I would do that in another fixer, it use will be limitted because of: https://github.com/sebastianbergmann/phpunit/issues/2074#issuecomment-183472385 (i.e. it will be removed from PHPUnit itself)

> Only fix sub classes from \PHPUnit_Framework_TestCase

I agree with @keradus here, there is no way solid figuring out if a class (or some include / trait) is part of a `PHPUnit_Framework_TestCase`-class inherence by looking at the tokens of that single file.

> Rename to PhpUnitExpectedExceptionFixer

only if your fixer also fixes the casing ;) (I would like that feature btw., but still think it is out of scope)

**Comment by keradus** [comment]
> Rename to `PhpUnitExpectedExceptionFixer`

as we don't modifying `->expectException` I would name it `PhpUnitFqcnAnnotationFixer`.
changing string passed to method is different thing and it should be separated fixer/PR.

**Comment by ro0NL** [comment]
:+1: lets keep things simple.

File: Symfony/CS/Tests/Fixer/Symfony/PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixerTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\Symfony;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/PhpUnitExpectedExceptionAnnotationWithRootNamespaceFixerTest.php]: add @internal annotation

**Comment by keradus** [comment]
raname and ready to merge ?

**Comment by ro0NL** [comment]
Worst idea ever.. but i think it's ok :)

**Comment by keradus** [comment]
> Worst idea ever..

the name? it was just a proposal to have it short and not connected to function call with same name that annotation.
propose better one ;)

**Comment by ro0NL** [comment]
I meant renaming via github :smile: 

This is ready :tada:

**Comment by keradus** [comment]
:+1:

**Comment by SpacePossum** [comment]
:+1: :ship: thanks @ro0NL

**Comment by keradus** [comment]
Thank you @ro0NL.

**Pull request merged**